app/controllers/spotlight/resources/upload_controller.rb

```diff
@@ -14,16 +14,16 @@ class UploadController < ApplicationController
     def create
       @resource.attributes = resource_params
 
-      if @resource.save
+      if @resource.save_and_commit
         flash[:notice] = t('spotlight.resources.upload.success')
         if params["add-and-continue"]
           redirect_to new_exhibit_resources_upload_path(@resource.exhibit)
         else
-          redirect_to admin_exhibit_catalog_index_path(@resource.exhibit)
+          redirect_to admin_exhibit_catalog_index_path(@resource.exhibit, sort: :timestamp)
```

Comment:
What would you think of changing https://github.com/sul-dlss/spotlight/blob/master/config/routes.rb#L28 to be `get 'admin', defaults: {sort: 'timestamp'}` instead?